src/main.rs

```diff
@@ -34,15 +34,15 @@ mod logger;
 use logger::Logger;
 
 const TORRENT_FILE: &str = "Charlie_Chaplin_Mabels_Strange_Predicament.avi.torrent";
-const CONNECTION_TIMEOUT: Duration = Duration::from_millis(750);
-const READ_TIMEOUT: Duration = Duration::from_millis(2000);
-const PROGRESS_WAIT_TIME: Duration = Duration::from_secs(15);
+const CONNECTION_TIMEOUT: Duration = Duration::from_millis(250);
+const READ_TIMEOUT: Duration = Duration::from_millis(1000);
+const PROGRESS_WAIT_TIME: Duration = Duration::from_secs(10);
 const THREADS_PER_PEER: u8 = 1;
-const MAX_IN_PROGRESS_REQUESTS_PER_CONNECTION: usize = 30;
+const MAX_IN_PROGRESS_REQUESTS_PER_CONNECTION: usize = 128;
```

Comment:
This appears to be the key lever to pull. I want to play around with this value and see what we can learn.